### ISSUE TITLE
Drop official 32-bit (legacy) x86 builds

### ISSUE THREAD

**Issue by SoniEx2** [kvirc/KVIrc#1842]
The modern world is (mostly) 64-bit. You should drop official 32-bit x86 builds (or at least make them a bit harder to get). At least for Linux. (Distros can provide their own 32-bit builds and anyone on bleeding edge should know how to compile it themselves.)

**Comment by staticfox** [comment]
Is there really any reason to?

**Comment by SoniEx2** [comment]
Any reason to keep supporting 32-bit x86?

I mean I don't think anyone's gonna be running modern KVIrc on embedded systems (only place where you might reasonably expect 32-bit x86 linux). The only 2 things I know that run 32-bit x86 are embedded systems (which literally have no 64-bit support) and shitty computer brands (which run 32-bit windows, so that's a reason to keep 32-bit x86 **windows** builds, but usually have 64-bit CPUs).

(Not mentioning Apple because there are no official Apple builds, and Apple recently dropped 32-bit support anyway.)

**Comment by wodim** [comment]
- Many people use 32-bit versions of Windows.
- Many people use 32-bit versions of Linux.

I mean, what in hell?

**Comment by SoniEx2** [comment]
I said modern. 64-bit x86 has been around for 10 years. That's plenty of transition time.

**Comment by DarthGandalf** [comment]
@SoniEx2 why do you care which builds do we build? Feel free to not use them...

**Comment by ctrlaltca** [comment]
I propose to drop builds with an odd version number, because they are odd anyway.

**Comment by un1versal** [comment]
> The modern world is (mostly) 64-bit. You should drop official 32-bit x86 builds (or at least make them a bit harder to get). At least for Linux. (Distros can provide their own 32-bit builds and anyone on bleeding edge should know how to compile it themselves.)

Words like _mostly_ and thoughts like "make it harder to get at least for Linux", makes me wonder **What the...**?

> I said modern. 64-bit x86 has been around for 10 years. That's plenty of transition time.

A few months back one of our users was having an issue with nightly 32-bit builds, but he never answered any questions about his system (none), and I naturally assumed it was old ass, ancient HW, and told him to upgrade so as he could run 64-bit builds. Not only did I get then and get now some stick for that from @staticfox but it turns out after weeks of hammering this, that he was merely running 32-bit version of Windows 7... to this day he refuses to even upgrade OS bitness.

Let me chuck chuck my wood into the fire right here. 
Some softwares I know and use are 32-bit only they work fine on 64-bit OS's they have no plans to build 64-bit anytime soon or maybe even ever!

Then you have those who abandoned 32-bit builds altogether for your kind of (somewhat faulty) reasoning and are asked by a very very small subset of users for 32-bit builds.

> (Not mentioning Apple because there are no official Apple builds, and Apple recently dropped 32-bit support anyway.)

You shouldn't compare apples and oranges anyway, KVirc is not an OS. We also are opensource Apple is not.

> Many people use 32-bit versions of Windows.
> 
> ```
> Many people use 32-bit versions of Linux.
> ```
> 
> I mean, what in hell?

Even though this is a non substantiated vague comment by wodim (where KVIrc is concerned,) I agree, but only because I know people who are actually poor, so they dont throw away old computers and buy shinny new ones every 5 years as they probably prefer to eat, wear clothes, and sleep under a roof,

I wonder how  Dr Stanley Milgram would approach this, but I think you would be disappointed to find out that your request in some way would cause our lovely, stubborn and poor users to suffer;
—you (and a lot of other people) are being (told) brain washed into believing (among other things) 32-bit is some horrible unsightly and worthy of being ripped out from the bosom of technology and relegated to a footnote in history and therefore your post actually reflecs this.

TL;DR: **Sorry** if I seemed harsh (not meant to), I wanted to type out something with my first cup of coffee and also give my **_\- 1**_

Enough minus views here, so closing this if noone else minds.

:

**Comment by wodim** [comment]
@SoniEx2 you sound like one of those ipv6 zealots.

**Comment by SoniEx2** [comment]
I have a 10 year old PC running arch linux x86_64. x86_64-compatible PCs are a lot more common than you think, and 99.9% of the people on 32-bit are either windows users (and I didn't tell you to drop 32-bit windows) or know how to compile stuff themselves.

**Comment by ctrlaltca** [comment]
Discussion pointless turning into a flame. locked.
